core/src/main/java/org/apache/accumulo/core/metadata/schema/LinkingIterator.java

```diff
@@ -132,7 +146,23 @@ public TabletMetadata next() {
 
       if (prevTablet == null) {
         if (tmp.sawPrevEndRow()) {
-          currTablet = tmp;
+
+          Text prevRow = tmp.getExtent().getPrevEndRow();
+          Text prevMetaRow = null;
+          if (prevRow != null) {
+            prevMetaRow = TabletsSection.getRow(tmp.getExtent().getTableId(), prevRow);
+          }
```

Comment:
Could make the code a little more easy to understand by reducing to use only a single variable:

```java
Text prevMetaRow = Optional.ofNullable(tmp.getExtent().getPrevEndRow()).map(prevRow -> TabletsSection.getRow(tmp.getExtent().getTableId(), prevRow).orElse(null);
```

Reply:
Or just call `getPrevEndRow()` twice in a single if statement.